src/build_queue.rs

```diff
@@ -188,6 +188,27 @@ impl BuildQueue {
     }
 }
 
+fn retry<T>(mut f: impl FnMut() -> Result<T>) -> Result<T> {
+    const MAX_ATTEMPTS: u8 = 3;
```

Comment:
If you're going to make `retry` reusable, I would expect this to be a function parameter.
(Not convinced this *should* be reusable if we only use it in one place, but no strong objection.)

Reply:
I made this an parameter